src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/search/SearchByNameAndPrisonNumber.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search
+
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.RecentArrival
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search.Searcher.Result
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search.Searcher.SearchStrategy
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search.Weights.Companion.CONSTANT
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search.Weights.Companion.EXACT_MATCH
+import uk.gov.justice.digital.hmpps.welcometoprison.model.arrivals.search.Weights.Companion.PARTIAL_MATCH
+
+class SearchByNameAndPrisonNumber : SearchStrategy<String, RecentArrival> {
+  private val fuzzyScorer = FuzzyScorer()
+
+  override fun evaluate(query: String, item: RecentArrival) = Result(item, calculateRelevance(query, item))
+
+  private fun calculateRelevance(query: String, item: RecentArrival): Int? {
+    val fields = item.splitToFields()
+    val terms = query.splitToTerms()
+    return when {
+      terms.isEmpty() -> CONSTANT
```

Comment:
no query param will already have been handled by the searcher, but need to handle empty string or when query is just whitespace